include/seqan3/argument_parser/exceptions.hpp

```diff
@@ -34,93 +45,83 @@ namespace seqan3
-class too_few_arguments : public parser_invalid_argument
+//!\brief Argument parser exception thrown when encountering unknown option.
+class unknown_option : public argument_parser_error
 {
 public:
     /*!\brief The constructor.
      * \param[in] s The error message.
      */
-    too_few_arguments(std::string const & s) : parser_invalid_argument(s) {}
+    unknown_option(std::string const & s) : argument_parser_error(s) {}
 };
 
-//!\brief Argument parser exception thrown when a required option is missing.
-class required_option_missing : public parser_invalid_argument
+//!\brief Argument parser exception thrown when too many arguments are provided.
+class too_many_arguments : public argument_parser_error
 {
 public:
     /*!\brief The constructor.
      * \param[in] s The error message.
      */
-    required_option_missing(std::string const & s) : parser_invalid_argument(s) {}
+    too_many_arguments(std::string const & s) : argument_parser_error(s) {}
 };
 
-//!\brief Argument parser exception thrown when a non-list option is declared multiple times.
-class option_declared_multiple_times : public parser_invalid_argument
+//!\brief Argument parser exception thrown when too few arguments are provided.
+class too_few_arguments : public argument_parser_error
 {
 public:
     /*!\brief The constructor.
      * \param[in] s The error message.
      */
-    option_declared_multiple_times(std::string const & s) : parser_invalid_argument(s) {}
+    too_few_arguments(std::string const & s) : argument_parser_error(s) {}
 };
 
-//!\brief Argument parser exception thrown when an argument could not be casted to the according type.
-class type_conversion_failed : public parser_invalid_argument
+//!\brief Argument parser exception thrown when a required option is missing.
+class required_option_missing : public argument_parser_error
 {
 public:
     /*!\brief The constructor.
      * \param[in] s The error message.
      */
-    type_conversion_failed(std::string const & s) : parser_invalid_argument(s) {}
+    required_option_missing(std::string const & s) : argument_parser_error(s) {}
 };
 
-//!\brief Argument parser exception thrown when an argument could not be casted to the according type.
-class overflow_error_on_conversion : public parser_invalid_argument
+//!\brief Argument parser exception thrown when a non-list option is declared multiple times.
+class option_declared_multiple_times : public argument_parser_error
 {
 public:
     /*!\brief The constructor.
      * \param[in] s The error message.
      */
-    overflow_error_on_conversion(std::string const & s) : parser_invalid_argument(s) {}
+    option_declared_multiple_times(std::string const & s) : argument_parser_error(s) {}
 };
 
 //!\brief Argument parser exception thrown when an argument could not be casted to the according type.
-class validation_failed : public parser_invalid_argument
+class validation_error : public argument_parser_error
```

Comment:
So if I get this right. When people use one of this exceptions they will get an compiler error because it is no longer present. When they used parser_invalid_argument they will get a deprecation message but also thrown exceptions are never catched and the program terminates. So which means we have two options. Either deprecate everything or let the compiler always fail when using old exceptions. I am tending for the former solution. Or am I missing something here?

Reply:
Alright, I will deprecate all the old names and we can remove them with 3.1